src/urbit/doc/dev/data-types.md

```diff
@@ -0,0 +1,506 @@
+# Data Types
+
+Chess’s data types are defined across two `/sur` files and at the top of the `/app` file.
+
+## /sur/chess
+Most types in `/sur/chess` come with a default value ([“bunt”](https://developers.urbit.org/reference/glossary/bunt)) specified with the [`$~`](https://developers.urbit.org/reference/hoon/rune/buc#-bucsig) rune, mostly for illustrative purposes. Only twice in the code is the bunt of any of these types actually used.
+
+### chess-side
+```
++$  chess-side
+  $~  %white
+  $?  %white
+      %black
+  ==
+```
+
+A chess player can have one of two `chess-side`s: white or black.
+
+### chess-piece-type
+```
++$  chess-piece-type
+  $~  %pawn
+  $?  %pawn
+      %knight
+      %bishop
+      %rook
+      %queen
+      %king
+  ==
+```
+
+A chess piece can have one of six `chess-piece-type`s.
+
+### chess-promotion
+```
++$  chess-promotion
+  $~  %queen
+  $?  %knight
+      %bishop
+      %rook
+      %queen
+  ==
+```
+
+A pawn can be promoted to one of four `chess-piece-type`s.
+
+### chess-piece
+```
+$~  [%white %pawn]
+  $:  chess-side
+      chess-piece-type
+  ==
+```
+
+A `chess-piece` is a cell of the piece’s `chess-side` and `chess-piece-type`.
+
+### chess-rank
+```
++$  chess-rank
+  $~  %1
+  ?(%1 %2 %3 %4 %5 %6 %7 %8)
+```
+
+A chessboard has eight rows (“ranks”), numbered with literals `%1` to `%8`.
+
+### chess-file
+```
++$  chess-file
+  $~  %a
+  ?(%a %b %c %d %e %f %g %h)
+```
+
+A chessboard has eight columns (“files”), labelled with terms `%a` to `%h`.
+
+### chess-square
+```
++$  chess-square
+  $~  [%a %1]
+  [chess-file chess-rank]
+```
+
+A `chess-square` is a cell of `chess-rank` and `chess-file`.
```

Comment:
This documentation doesn't add additional context beyond what's available from just looking at the type definitions. It's basically explaining the rules of chess, which we should avoid.